consensus/clique/snapshot.go

```diff
@@ -308,5 +309,11 @@ func (s *Snapshot) inturn(number uint64, signer common.Address) bool {
 	for offset < len(signers) && signers[offset] != signer {
 		offset++
 	}
+
+	if len(signers) == 0 {
+		log.Warn("no signer. len(signers)==0")
```

Comment:
The warning message should maybe say something like `There are no signers for this chain, they have voted themselves out; this chain can never resume operation`
Otherwise I agree that returning `false` is the technically correct thing to do, but I would suspect other failures lurks behind this one.

Reply:
Btw should we allow empty signer happen? If so, no block will be produced, no new signer can be added. 
I guess we should add more restriction for `removeSigner` proposal, it there is only one signer left, reject it.

Reply:
@rjl493456442  Yes, I think so too. You can add restrictions to the "validVote" function, not vote for the last siger.

Reply:
I submitted it again in my Fork.

SHA-1: c15e5b0c73a3ef840ff67f7eb1932072319ac687
* If only the last signer is left, voting for it is invalid